geowebcache/core/src/test/java/org/geowebcache/mime/ImageMimeTest.java

```diff
@@ -114,4 +120,17 @@ public void testIsCompatileFailsOnXMLMime() {
         assertFalse("PNG mime type should not be compatible with XML error.",
             pngMimeType.isCompatible("application/vnd.ogc.se_xml"));
     }
+
+    @Test
+    public void test4BitPNG() throws IOException, URISyntaxException {
+        URL url = this.getClass().getResource("/images/4bit.png");
+        RenderedImage tile = ImageIO.read(new File(url.toURI()));
+        ImageWriter writer = ImageMime.png8.getImageWriter(tile);
+        //Not a great test, but make sure our writer is the pure java one, not the native version
+        //since the native version can't handle 4bit pngs.
```

Comment:
Agreed. This is probably the best that's practical, even if it's far from ideal.  All I can think of as an alternative would be wrapping the `javax.imageio.ImageIO.getImageWritersByFormatName(internalName);` call in a method that could be overridden for testing and then providing mocks, which still has the problem of providing a mock for a class that may or may not exist where the specific class name is the important detail.